### PR TITLE
Add replica parameter and shard number option list

### PR DESCRIPTION
**Pull request by xluo-aws**
Signed-off-by: Xuesong Luo <lxuesong@amazon.com>

### Description
[Add replica as optional parameter to split index; Provide valid shard number options for user to choose from]

### Issues Resolved
[List any issues this PR will resolve https://github.com/opensearch-project/index-management-dashboards-plugin/issues/393]

### Check List
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by SuZhou-Joe — DISMISSED** [review]

File: public/pages/Indices/components/SplitIndexFlyout/SplitIndexFlyout.tsx [hunk 1]
```diff
@@ -43,21 +43,55 @@ export default class SplitIndexFlyout extends Component<SplitIndexProps> {
...
+    const shardsSelectOptions = [];
+    if (sourceShards == 1) {
+      for (let i = 2; i <= 1024; i++) {
+        shardsSelectOptions.push({
+          value: i.toString(),
+          text: i,
+        });
+      }
+    } else {
+      const SHARDS_HARD_LIMIT = 1024 / 2;
```
Comment (Hailong-am) [inline public/pages/Indices/components/SplitIndexFlyout/SplitIndexFlyout.tsx:73]:
```suggestion
      let shard = sourceShards * 2;
      while (shard < SHARD_HARD_LIMIT) {
       shardsSelectOptions.push({
          value: shard.toString(),
          text: shard,
      });
      shard += sourceShards;
     }
}
```

File: public/pages/Indices/components/SplitIndexFlyout/SplitIndexFlyout.tsx [hunk 2]
```diff
@@ -36,28 +36,60 @@ interface SplitIndexProps {
...
+    if (sourceShards == 1) {
+      for (let i = 2; i <= 1024; i++) {
+        shardsSelectOptions.push({
+          label: i.toString(),
+        });
+      }
+    } else {
+      const SHARDS_HARD_LIMIT = 1024 / 2;
+      let shardsLimit = sourceShards;
+      for (let i = 1; shardsLimit <= SHARDS_HARD_LIMIT; i++) {
```
Comment (Hailong-am) [inline public/pages/Indices/components/SplitIndexFlyout/SplitIndexFlyout.tsx:74]:
```suggestion
      for (;shardsLimit <= SHARDS_HARD_LIMIT;) {
```

File: public/pages/Indices/components/SplitIndexFlyout/SplitIndexFlyout.tsx [hunk 3]
```diff
@@ -36,28 +36,60 @@ interface SplitIndexProps {
...
+
+      for (let i = sourceShards * 2; i <= shardsLimit; ) {
+        if (shardsLimit % i == 0) {
+          shardsSelectOptions.push({
+            label: i.toString(),
+          });
+          i = i * 2;
+        }
+      }
+    }
```
Comment (Hailong-am) [inline public/pages/Indices/components/SplitIndexFlyout/SplitIndexFlyout.tsx:86]:
```suggestion
    const SHARDS_HARD_LIMIT = 1024 / 2;
    
    for (let i = sourceShards * 2; i <= SHARDS_HARD_LIMIT; ) {
      shardsSelectOptions.push({
        label: i.toString(),
      });
      i = i * 2;    
    }
```

**Review by Hailong-am — DISMISSED** [review]

File: public/pages/Indices/components/SplitIndexFlyout/SplitIndexFlyout.tsx (hunk 3 above)
Comment (xluo-aws) [reply public/pages/Indices/components/SplitIndexFlyout/SplitIndexFlyout.tsx:86]: Will make the change

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**